lib/ProductOpener/Tags.pm

```diff
@@ -1702,6 +1704,188 @@ sub build_tags_taxonomy($$$) {
+			# Compute a list of the requested inherited properties,
+			# as we will populate them directly
+			if ($field =~  /^inherited:(.*):(\w\w)$/) {
+				my $prop = $1;
+				my $lc = $2;
+				push @properties, [$prop, $lc];
+			}
+			# Compute a hash of the other requested fields
+			# as we will go through all existing properties
+			else {
+				$fields_ref->{$field} = 1;
+			}
+		}
+	}
+
+	my $taxonomy_ref = {};
+
+	while (my $tagid = shift @tags) {
+
+		if (defined $direct_parents{$tagtype}{$tagid}) {
+			
+			$taxonomy_ref->{$tagid}{parents} = [];
+			foreach my $parentid (sort keys %{$direct_parents{$tagtype}{$tagid}}) {
+				if ((not defined $fields_ref) or (defined $fields_ref->{parents})) {
+					exists $taxonomy_ref->{$tagid}{parents} or $taxonomy_ref->{$tagid}{parents} = [];
+					push @{ $taxonomy_ref->{$tagid}{parents} }, $parentid;
+				}
+				# Include parents if the tag is one of the initially requested tags
+				# so that we don't also add parents of parents
+				if (($options_ref->{include_parents}) and ($requested_tags{$tagid})) {
+					push @tags, $parentid;
+				}
+			}
+		}
+
+		if (defined $direct_children{$tagtype}{$tagid}) {
+			
+			foreach my $childid (sort keys %{$direct_children{$tagtype}{$tagid}}) {
+				if ((not defined $fields_ref) or (defined $fields_ref->{children})) {
+					exists $taxonomy_ref->{$tagid}{children} or $taxonomy_ref->{$tagid}{children} = [];
+					push @{$taxonomy_ref->{$tagid}{children}}, $childid;
+				}
+				if (($options_ref->{include_children}) and ($requested_tags{$tagid})) {
+					push @tags, $childid;
+				}
+			}
+		}
+
+		if (((not defined $fields_ref) or (defined $fields_ref->{name}))
+			and (defined $translations_to{$tagtype}{$tagid})) {
+
+			$taxonomy_ref->{$tagid} = {name => {}};
+
+			foreach my $lc (@{$lcs_ref}) {
+
+				if (defined $translations_to{$tagtype}{$tagid}{$lc}) {
+					$taxonomy_ref->{$tagid}{name}{$lc} = $translations_to{$tagtype}{$tagid}{$lc};
+				}
+
+				my $lc_tagid = get_string_id_for_lang($lc, $translations_to{$tagtype}{$tagid}{$lc});
+
+				if (defined $synonyms_for{$tagtype}{$lc}{$lc_tagid}) {
+
+					# additives has e-number as their name, and the first synonym is the additive name
+					if (($tagtype =~ /^additives(|_prev|_next|_debug)$/) and (defined $synonyms_for{$tagtype}{$lc}{$lc_tagid}[1])) {
+						$taxonomy_ref->{$tagid}{name}{$lc} .= " - " . $synonyms_for{$tagtype}{$lc}{$lc_tagid}[1];
+					}
+
+					# add synonyms to the full taxonomy
+					if (((not defined $fields_ref) or (defined $fields_ref->{synonyms}))
+						and (defined $synonyms_for{$tagtype}{$lc}{$lc_tagid})) {
+						(defined $taxonomy_ref->{$tagid}{synonyms}) or $taxonomy_ref->{$tagid}{synonyms} = {};
+						$taxonomy_ref->{$tagid}{synonyms}{$lc} = $synonyms_for{$tagtype}{$lc}{$lc_tagid};
+					}
+				}
+			}
+		}
+
+		if (defined $properties{$tagtype}{$tagid}) {
+
+			foreach my $prop_lc (keys %{$properties{$tagtype}{$tagid}}) {
+				
+				if ($prop_lc =~ /^(.*):(\w\w)$/) {
```

Comment:
it would be interesting to know why we do nothing if prop_lc does not match the pattern.

Reply:
It always match the pattern. I guess we could remove the if and do it differently.

Something like this should work:

# properties are of the form [property_name]:[2 letter language code]
my ($prop, $lc) = split(/:/, $prop_lc)